src/components/Whiteboard/ElementBehaviors/Resizable/utils.ts

```diff
@@ -88,7 +172,8 @@ export function calculateDimensions(
   viewportHeight: number,
   forceLockAspectRatio: boolean = false,
   gridCellSize?: number,
-): Dimensions {
+  startingPoints?: Point[],
+): Dimensions & Points {
```

Comment:
This change makes this API difficult to understand. The `startingPoints` that are available for both `kind`s `line` and `polyline` are passed for the line only and later on in this method if `startingPoints` are defined they are used to run specific method for the line calculation: `calculateLineDragDimensions` skipping the rest of `calculateDimensions`. The `& Points` of `Dimensions & Points` has '.points' that is an optional member (that is in contrast to all `Dimentions` members) that is set only for the `line` calculation, so making this method to return objects with different structure for both line and polyline. Would be good to improve this API that it is easier to understand how it works and what it returns for different types of elements that support resizing.